src/server/stream_family.cc

```diff
@@ -907,6 +907,144 @@ OpResult<pair<stream*, streamCG*>> FindGroup(const OpArgs& op_args, string_view
   return res;
 }
 
+constexpr uint8_t kClaimForce = 1 << 0;
+constexpr uint8_t kClaimJustID = 1 << 1;
+
+struct ClaimOpts {
+  string_view group;
+  string_view consumer;
+  int64 min_idle_time;
+  ParsedStreamId lastid;
+  int64 deliverytime = -1;
+  int retry = -1;
+  uint8_t flags = 0;
+};
+
+struct ClaimInfo {
+  bool justid = false;
+  vector<streamID> ids;
+  RecordVec records;
+};
+
+// XCLAIM key group consumer min-idle-time id
+OpResult<ClaimInfo> OpClaim(const OpArgs& op_args, string_view key, const ClaimOpts& opts,
+                            absl::Span<streamID> ids) {
+  OpResult<pair<stream*, streamCG*>> cgr_res = FindGroup(op_args, key, opts.group);
+  if (!cgr_res)
+    return cgr_res.status();
+  stream* s = cgr_res->first;
+  streamCG* scg = cgr_res->second;
+  if (!scg) {
+    return OpStatus::SKIPPED;
+  }
+  streamConsumer* consumer = nullptr;
+  // TODO: Use GetCurrentTimeMs() instead?
+  auto now = GetCurrentTimeMs();
+  ClaimInfo result;
+  result.justid = (opts.flags & kClaimJustID);
+  auto AppendResultItem = [&](stream* s, streamID id) {
+    int64_t numfields;
+    if (result.justid) {
+      result.ids.push_back(id);
+    } else {
+      streamIterator it;
+      streamID cid;
+      streamIteratorStart(&it, s, &id, &id, 0);
+      while (streamIteratorGetID(&it, &cid, &numfields)) {
+        Record rec;
+        rec.id = cid;
+        rec.kv_arr.reserve(numfields);
+
+        /* Emit the field-value pairs. */
+        while (numfields--) {
+          unsigned char *key, *value;
+          int64_t key_len, value_len;
+          streamIteratorGetField(&it, &key, &value, &key_len, &value_len);
+          string skey(reinterpret_cast<char*>(key), key_len);
+          string sval(reinterpret_cast<char*>(value), value_len);
+
+          rec.kv_arr.emplace_back(std::move(skey), std::move(sval));
+        }
+        result.records.push_back(std::move(rec));
+      }
+      streamIteratorStop(&it);
+    }
+  };
+
+  for (streamID id : ids) {
+    uint8_t buf[sizeof(streamID)];
```

Comment:
It's a good practice to use `std::array<uint8_t, sizeof(streamID)>` instead. It provides better safety measures in the general case.